lib/active_resource/base.rb

```diff
@@ -1443,6 +1443,8 @@ def encode(options = {})
     #   my_branch.name # => "Wilson Road"
     def reload
       self.load(self.class.find(to_param, params: @prefix_options).attributes, false, true)
+    rescue => exception
+      rescue_with_handler(exception) || raise
```

Comment:
If https://github.com/rails/activeresource/pull/414 were merged, this change could be removed in favor of a matching declaration in `ActiveResource::Rescuable`:

```ruby
around_reload :rescue_exceptions
```